### PR TITLE
New config fields and bugfixes for LED config restore and #294

### PR DESCRIPTION
**Pull request by skaman82**
* Adjustments Tab: Added LEVEL and other yet missing PIDs
* Receiver Tab: Added DEADBAND and YAW DEADBAND fields
* Pid Tab: PIDs can be reset to DEFAULTS if you want to start fresh for tuning
* Config Tab: Sensor alignment fields for ACC, MAG and GYRO - plus supporting images for board orientation

* Bugfix1: Restoring LED configuration from backup file works again
* Bugfix2: Restoring BACKUP functionality for older firmware versions (got broken after the merge of @ProDrone Failsafe Tab)

@hydra This requires supporting firmware PR from @NightHawk32. 
Like I mentioned, this is the last multi-PR. Will do more focused PRs in the future. 
Will do a CSS cleanup PR next and remove all the inline CSS.

### CODE REVIEW

**Comment by skaman82** [comment]
Found a bug. Try to resolve it with @NightHawk32 this evening.

File: js/backup_restore.js [hunk 1]
```diff
@@ -714,7 +752,11 @@ function configuration_restore(callback) {
             }
 
             function send_led_strip_config() {
-                MSP.sendLedStripConfig(reboot);
+                MSP.sendLedStripConfig(save_to_eeprom);
+            }
+            
+            function save_to_eeprom() {
+                MSP.send_message(MSP_codes.MSP_EEPROM_WRITE, false, false, reboot);
```
Comment (ProDrone) [inline js/backup_restore.js]: eeprom is now saved twice in a row... line 704 should only be a call to `send_led_strip_config()`

**Comment by skaman82** [comment]
nice catch, thank you

File: js/backup_restore.js (hunk 1 above)
Comment (NightHawk32) [inline js/backup_restore.js]: Thanks for that, you are completely right!

**Comment by ProDrone** [comment]
And while i'm at it...

Look at this: https://github.com/cleanflight/cleanflight-configurator/blob/development/js/msp.js#L1403-L1405
This is an error in logic i think. What happens AFTER, if onCompleteCallback(); ever gets called?

I think this:
~~~C
if (MODE_RANGES.length == 0) {
    onCompleteCallback();
}
send_next_mode_range();
.
.

~~~
Should be changed to this:
~~~C
if (MODE_RANGES.length == 0) {
    onCompleteCallback();
} else {
    send_next_mode_range();
}
.
.
~~~
This pattern is repeated several times.

**Comment by skaman82** [comment]
@hydra fixed. this is ready now

File: js/backup_restore.js (hunk 1 above)
Comment (skaman82) [inline js/backup_restore.js]: fixed in https://github.com/skaman82/cleanflight-configurator/commit/5160855b93b4d7f19bfbdada31513980ef3ffa8d

**Comment by skaman82** [comment]
supporting firmware PR for this is here: https://github.com/cleanflight/cleanflight/pull/1584

**Comment by ProDrone** [comment]
@skaman82 What i meant with "This pattern is repeated several times." is, that the same bug appears in all places that use the same pattern.
It is now only fixed in one function: `MSP.sendModeRanges`.
The same error also appears in functions: 
~~~
MSP.sendLedStripConfig
MSP.sendAdjustmentRanges
MSP.sendServoConfigurations
~~~
Please also apply the same modification to these.

**Comment by skaman82** [comment]
@ProDrone Done in https://github.com/skaman82/cleanflight-configurator/commit/70b35f22c9196c0eb2f69606df075cb6b17b7746 I usually never touch the JS because I don't know what I'm doing. Thats why @NightHawk32 does all the JS stuff usually. But I think I get this correctly this time. :)

**Comment by ProDrone** [comment]
Looks good to me, except for the indention of the function calls below
~~~C
} else {
~~~

**Comment by skaman82** [comment]
fixed indention

**Comment by hydra** [comment]
this needs rebasing against current development branch

**Comment by skaman82** [comment]
@hydra Done!

**Comment by skaman82** [comment]
@ProDrone also done

**Comment by skaman82** [comment]
@ProDrone Hmmm when I do this the LED config restore don't work anymore

**Comment by skaman82** [comment]
@hydra ready for merging. The corresponding firmware PR is here: https://github.com/cleanflight/cleanflight/pull/1630

**Comment by hydra** [comment]
https://github.com/cleanflight/cleanflight/pull/1630 is now merged, this is unblocked.

Please @skaman82  and @ProDrone test against latest development firmware and report back.

**Comment by hydra** [comment]
@NightHawk32 please also test using latest firmware and report back.

**Comment by skaman82** [comment]
Just tested this with 2642_4843a73_travis_cleanflight_NAZE.hex build. Everything works as expected. Tested the new sensor alignment and deadband fields, PID reset, new adjustments and save and restore.

**Comment by NightHawk32** [comment]
Also did some testing and couldnt findy any issues. Tried to restore some old configs and migration worked quite well.
I used the the same version as @skaman82.

**Comment by skaman82** [comment]
@hydra added some CSS cleanup and removed all inline CSS in HTML files. All tested and ready to go :)

**Comment by hydra** [comment]
roll and yaw fixed - select boxes with labels

@NightHawk32 @skaman82 see photo

**Comment by hydra** [comment]
roll and yaw fixed - select boxes with labels 2

fixed up a bit more.

**Comment by hydra** [comment]
ok, merging now with the fixes above.  if there's more fixes we will addres them in seperate issues.

**Comment by skaman82** [comment]
+1

Von meinem iPhone gesendet

**Comment by skaman82** [comment]
@hydra the new sensor alignment labels are way too long to fit in. See: 
bildschirmfoto 2016-01-07 um 02 52 25
I will shorten them again. Ok?

**Pull request merged**